src/fairseq2/cluster.py

```diff
@@ -97,81 +95,15 @@ def __init__(self, cluster: str) -> None:
 @final
 class SlurmHandler(ClusterHandler):
     def __init__(self, env: Environment) -> None:
-        self._job_id: int | None = None
-        self._env = env
+        self._job = clusterscope.get_job()
```

Comment:
Unfortunately this is problematic. First of all `get_job()` is a heavy call when `is_slurm_job()` is true including running scontrol as a subprocess which we should not really call in a constructor. This constructor is also executed way early in the process lifetime while we build the dependency graph. So if anything fails in `get_job()`, the user will have no clue what is going on even if they don't use our cluster API at all.